core/src/data/value/binary_op/i128.rs

```diff
@@ -0,0 +1,660 @@
+        assert_eq!(
+            base.try_add(&Decimal(Decimal::ONE)),
+            Ok(Decimal(Decimal::TWO))
+        );
+
+        assert_eq!(
+            base.try_add(&Bool(true)),
+            Err(ValueError::NonNumericMathOperation {
+                lhs: I128(base),
+                operator: NumericBinaryOperator::Add,
+                rhs: Bool(true)
+            }
+            .into())
+        );
+    }
+
+    #[test]
+    fn try_subtract() {
+        let base = 1_i128;
+
+        assert_eq!(base.try_subtract(&I8(1)), Ok(I8(0)));
+        assert_eq!(base.try_subtract(&I64(1)), Ok(I64(0)));
+        assert_eq!(base.try_subtract(&I128(1)), Ok(I128(0)));
+
+        assert!(
+            matches!(base.try_subtract(&F64(1.0)), Ok(F64(x)) if (x - 0.0).abs() < f64::EPSILON )
+        );
+
+        assert_eq!(
+            base.try_subtract(&Decimal(Decimal::ONE)),
+            Ok(Decimal(Decimal::ZERO))
+        );
+
+        assert_eq!(
+            base.try_subtract(&Bool(true)),
+            Err(ValueError::NonNumericMathOperation {
+                lhs: I128(base),
+                operator: NumericBinaryOperator::Subtract,
+                rhs: Bool(true)
+            }
+            .into())
+        );
+    }
+
+    #[test]
+    fn try_multiply() {
+        let base = 3_i128;
+
+        // 3 * 2 = 6
+        assert_eq!(base.try_multiply(&I8(2)), Ok(I8(6)));
+        assert_eq!(base.try_multiply(&I64(2)), Ok(I64(6)));
+        assert_eq!(base.try_multiply(&I128(2)), Ok(I128(6)));
+
+        assert_eq!(base.try_multiply(&I8(-1)), Ok(I8(-3)));
+        assert_eq!(base.try_multiply(&I64(-1)), Ok(I64(-3)));
+        assert_eq!(base.try_multiply(&I128(-1)), Ok(I128(-3)));
+
+        assert!(
+            matches!(base.try_multiply(&F64(1.0)), Ok(F64(x)) if (x - 3.0).abs() < f64::EPSILON )
+        );
+
+        let _result: Decimal = Decimal::from(3);
+        assert_eq!(
+            base.try_multiply(&Decimal(Decimal::ONE)),
+            Ok(Decimal(_result))
+        );
+
+        assert_eq!(
+            base.try_multiply(&Bool(true)),
+            Err(ValueError::NonNumericMathOperation {
+                lhs: I128(base),
+                operator: NumericBinaryOperator::Multiply,
+                rhs: Bool(true)
+            }
+            .into())
+        );
+    }
+
+    #[test]
+    fn try_divide() {
+        let base = 6_i128;
+
+        // 6/2 = 3
+        assert_eq!(base.try_divide(&I8(2)), Ok(I8(3)));
+        assert_eq!(base.try_divide(&I64(2)), Ok(I64(3)));
+        assert_eq!(base.try_divide(&I128(2)), Ok(I128(3)));
+
+        // 6/-6 = -1
+        assert_eq!(base.try_divide(&I8(-6)), Ok(I8(-1)));
+        assert_eq!(base.try_divide(&I64(-6)), Ok(I64(-1)));
+        assert_eq!(base.try_divide(&I128(-6)), Ok(I128(-1)));
+
+        assert!(
+            matches!(base.try_divide(&F64(1.0)), Ok(F64(x)) if (x - 6.0).abs() < f64::EPSILON )
+        );
+
+        let _decimal_result = Decimal::from(base);
```

Comment:
Variables prefixed with `_` in Rust usually indicate "unused" .
<https://stackoverflow.com/questions/48361537/why-do-underscore-prefixed-variables-exist>

Additionally, if used in the same way as `let _`, it will be dropped on the next line.
playground: <https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6aac0bfc6812eabdfe2e1f2ca0eb29e6>

Reply:
bad habit from my python days...   Thank you..